midi_app_controller/gui/_tests/test_binds_editor.py

```diff
@@ -0,0 +1,206 @@
+        button_binds=button_binds_sample,
+        knob_binds=knob_binds_sample,
+    )
+
+
+@pytest.fixture
+def button_binds_fixture(qtbot, controller_sample, button_binds_sample):
+    actions = ["play_action", "stop_action"]
+    widget = ButtonBinds(controller_sample.buttons, button_binds_sample, actions)
+    qtbot.addWidget(widget)
+    return widget
+
+
+@pytest.fixture
+def knob_binds_fixture(qtbot, controller_sample, knob_binds_sample):
+    actions = ["volume_up", "volume_down", "zoom_in", "zoom_out"]
+    widget = KnobBinds(controller_sample.knobs, knob_binds_sample, actions)
+    qtbot.addWidget(widget)
+    return widget
+
+
+@pytest.fixture
+def binds_editor_fixture_basic(qtbot, controller_sample, binds_sample):
+    actions = ["play_action", "volume_up", "volume_down", "zoom_in", "zoom_out"]
+    widget = BindsEditor(
+        controller_sample, binds_sample, actions, save_binds=lambda x, y: None
+    )
+    qtbot.addWidget(widget)
+    widget.show()
+    return widget
+
+
+@pytest.fixture
+def binds_editor_fixture(qtbot, controller_sample, binds_sample):
+    with patch.object(
+        BindsEditor, "_save_and_exit"
+    ) as mock_save_and_exit, patch.object(BindsEditor, "_exit") as mock_exit:
+        actions = ["play_action", "volume_up", "volume_down", "zoom_in", "zoom_out"]
+        widget = BindsEditor(
+            controller_sample, binds_sample, actions, save_binds=lambda x, y: None
+        )
+        qtbot.addWidget(widget)
+        widget.show()
+        yield widget, mock_save_and_exit, mock_exit
+
+
+def test_button_binds_initialization(button_binds_fixture):
+    assert len(button_binds_fixture.button_combos) == 2
+
+
+def test_get_button_binds(button_binds_fixture):
+    binds = button_binds_fixture.get_binds()
+    assert len(binds) == 2
+    assert binds[0].action_id == "play_action"
+    assert binds[1].action_id == "stop_action"
+
+
+def test_knob_binds_initialization(knob_binds_fixture):
+    assert len(knob_binds_fixture.knob_combos) == 2
+
+
+def test_get_knob_binds(knob_binds_fixture):
+    binds = knob_binds_fixture.get_binds()
+    assert len(binds) == 2
+    assert binds[0].action_id_increase == "volume_up"
+    assert binds[1].action_id_decrease == "zoom_out"
+
+
+def test_binds_editor_switch_views(binds_editor_fixture_basic, qtbot):
+    widget = binds_editor_fixture_basic
+    QTest.mouseClick(widget.buttons_radio, Qt.LeftButton)
+    assert widget.buttons_widget.isVisible()
+    assert not widget.knobs_widget.isVisible()
+
+    QTest.mouseClick(widget.knobs_radio, Qt.LeftButton)
+    assert widget.knobs_widget.isVisible()
+    assert not widget.buttons_widget.isVisible()
+
+
+def test_binds_editor_save_and_exit(binds_editor_fixture, qtbot):
+    widget, mock_save_and_exit, _ = binds_editor_fixture
+    all_buttons = widget.findChildren(QPushButton)
+    save_and_exit_button = next(
+        (btn for btn in all_buttons if btn.text() == "Save and exit"), None
+    )
```

Comment:
Wouldn't making the button an attribute of `BindsEditor` be easier and more readable?